### PR TITLE
ddl: init global vars from system tables before start domain

### LINKED ISSUE #54860: Flacky test because of the location mismatch
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run any tests with testkit, it will have chance to trigger this issue. However, it's not easy to reproduce.

### 2. What did you expect to see? (Required)

```
panic: assert failed, location mismatch, ctxLoc: Local, varsLoc: Local, stmtLoc: Local [recovered]
	panic: assert failed, location mismatch, ctxLoc: Local, varsLoc: Local, stmtLoc: Local

goroutine 1691 [running]:
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile.func1()
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/executor/compiler.go:57 +0x491
panic({0x5a741a0?, 0xc00753c5d0?})
	/usr/lib/go/src/runtime/panic.go:770 +0x132
github.com/pingcap/tidb/pkg/util/intest.doPanic(...)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/util/intest/assert.go:59
github.com/pingcap/tidb/pkg/util/intest.Assert(...)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/util/intest/assert.go:27
github.com/pingcap/tidb/pkg/expression/context.AssertLocationWithSessionVars(0xa5a4be0, 0xc0063f2008)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/context/context.go:232 +0x1f9
github.com/pingcap/tidb/pkg/expression/contextsession.(*SessionEvalContext).TypeCtx(0xc0071943f0)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/contextsession/sessionctx.go:209 +0xbd
github.com/pingcap/tidb/pkg/expression.typeCtx(...)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/context.go:57
github.com/pingcap/tidb/pkg/expression.(*Constant).Equal(0xc0063f8240, {0x70d3798, 0xc0071943f0}, {0x7116908?, 0xc0063f8240})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/constant.go:437 +0xcb
github.com/pingcap/tidb/pkg/expression.(*baseBuiltinFunc).equal(0xc007194630, {0x70d3798, 0xc0071943f0}, {0x70ef260?, 0xc007194630?})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/builtin.go:400 +0xc3
github.com/pingcap/tidb/pkg/expression.(*ScalarFunction).Equal(0xc007179ab0, {0x70d3798, 0xc0071943f0}, {0x7116a70?, 0xc007179ab0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/expression/scalar_function.go:372 +0xc2
github.com/pingcap/tidb/pkg/planner/cardinality.getMaskAndRanges({0x70d9520, 0xc00753c0f0}, {0xc00753c550, 0x1, 0x1}, 0x90?, {0x0?, 0x63bc201?, 0xc001aab428?}, 0x0, ...)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/cardinality/selectivity.go:730 +0x4c6
github.com/pingcap/tidb/pkg/planner/cardinality.Selectivity({0x70d9520, 0xc00753c0f0}, 0xc007622a20, {0xc00753c4b0, 0x1, 0x1}, {0xc00753c1d0, 0x1, 0x1?})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/cardinality/selectivity.go:130 +0xbf1
github.com/pingcap/tidb/pkg/planner/core.(*DataSource).deriveStatsByFilter(0xc001ccc588, {0xc00753c4b0, 0x1, 0x1}, {0xc00753c1d0, 0x1, 0x2})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/stats.go:197 +0x159
github.com/pingcap/tidb/pkg/planner/core.(*DataSource).DeriveStats(0xc001ccc588, {0x0?, 0xc001ccc588?, 0xc006e303c0?}, 0xc001aac150?, {0x1e704fb?, 0x0?, 0xc001aac118?}, {0x0, 0x0, ...})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/stats.go:400 +0x59e
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0xc001ccc5a8, {0x0, 0x0, 0x0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/operator/logicalop/base_logical_plan.go:188 +0x29b
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0xc006e303c0, {0x0, 0x0, 0x0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x16f
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0xc0065202a0, {0x0, 0x0, 0x0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x16f
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0xc006e302f0, {0x0, 0x0, 0x0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x16f
github.com/pingcap/tidb/pkg/planner/core.physicalOptimize({0x71124c0, 0xc006e302d0}, 0xc007310240)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/optimizer.go:1046 +0x12d
github.com/pingcap/tidb/pkg/planner/core.doOptimize({0x708f408, 0xc00a26e410}, {0x70d9520, 0xc00753c0f0}, 0xf134a, {0x71124c0, 0xc006e302d0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/optimizer.go:308 +0x128
github.com/pingcap/tidb/pkg/planner/core.DoOptimize({0x708f408, 0xc00a26e410}, {0x70d9520, 0xc00753c0f0}, 0xf134a, {0x71124c0, 0xc006e302d0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/core/optimizer.go:355 +0x133
github.com/pingcap/tidb/pkg/planner.optimize({0x708f408, 0xc00a26e410}, {0x70d9520, 0xc00753c0f0}, {0x709e2d0, 0xc0085b3560}, {0x70de2d0, 0xc0070be600})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/optimize.go:525 +0xa16
github.com/pingcap/tidb/pkg/planner.Optimize({0x708f408, 0xc00a26e410}, {0x711b948, 0xc00741e508}, {0x709e2d0, 0xc0085b3560}, {0x70de2d0, 0xc0070be600})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/planner/optimize.go:356 +0x1937
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile(0xc0069a9a48, {0x708f408?, 0xc00a26e410?}, {0x70a81e0, 0xc0085b3560})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/executor/compiler.go:99 +0x48e
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt(0xc00741e508, {0x708f408?, 0xc00a26e410?}, {0x70a81e0, 0xc0085b3560})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/session/session.go:2098 +0x89a
github.com/pingcap/tidb/pkg/session.(*session).ExecuteInternal(0xc00741e508, {0x708f408, 0xc00a26e410}, {0xc006ede040, 0x3c}, {0x0, 0x0, 0x0})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/session/session.go:1523 +0x245
github.com/pingcap/tidb/pkg/ddl/session.(*Session).Execute(0xc00753c100, {0x708f408, 0xc00a26e410}, {0xc006ede040, 0x3c}, {0x63de7d9, 0xe})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/session/session.go:85 +0x1d0
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).GetMinJobID.func1(0xc00753c100)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/systable/manager.go:120 +0x9b
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).withNewSession(0xc002016348, 0xc00141adf8)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/systable/manager.go:71 +0xd1
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).GetMinJobID(0xc00a64c1e0?, {0x708f408?, 0xc00a26e410?}, 0xc009276d80?)
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/systable/manager.go:118 +0x66
github.com/pingcap/tidb/pkg/ddl/systable.(*MinJobIDRefresher).refresh(0xc00a0e5c98, {0x708f408?, 0xc00a26e410?})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/systable/min_job_id.go:66 +0x4b
github.com/pingcap/tidb/pkg/ddl/systable.(*MinJobIDRefresher).Start(0xc00a0e5c98, {0x708f408, 0xc00a26e410})
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/systable/min_job_id.go:54 +0x3d
github.com/pingcap/tidb/pkg/ddl.(*ddl).Start.func3()
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/ddl/ddl.go:869 +0x29
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run.func1()
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/util/wait_group_wrapper.go:157 +0x52
created by github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run in goroutine 87
	/home/yangkeao/Project/github.com/YangKeao/tidb/master/pkg/util/wait_group_wrapper.go:155 +0x73
FAIL	github.com/pingcap/tidb/pkg/executor	0.554s
FAIL
```

It's quite strange. The logic of `AssertLocationWithSessionVars` shows the `ctxLocStr` is different with `varsLocStr` (or `stmtLocStr`). But they are the same in the log:

```go
// AssertLocationWithSessionVars asserts the location in the context and session variables are the same.
// It is only used for testing.
func AssertLocationWithSessionVars(ctxLoc *time.Location, vars *variable.SessionVars) {
	ctxLocStr := ctxLoc.String()
	varsLocStr := vars.Location().String()
	stmtLocStr := vars.StmtCtx.TimeZone().String()
	intest.Assert(ctxLocStr == varsLocStr && ctxLocStr == stmtLocStr,
		"location mismatch, ctxLoc: %s, varsLoc: %s, stmtLoc: %s",
		ctxLoc.String(), ctxLocStr, stmtLocStr,
	)
}
```

### 3. What did you see instead (Required)

Test successfully. 

### 4. What is your TiDB version? (Required)

```
master-04e5ff999ac117552e144070a8ac440484fc44da
```

### LINKED ISSUE #54868: some inTest assert panic comes from ddl/systable/min_job_id when testing
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
panic: assert failed, location mismatch, ctxLoc: Local, varsLoc: Local, stmtLoc: Local [recovered]
	panic: assert failed, location mismatch, ctxLoc: Local, varsLoc: Local, stmtLoc: Local

goroutine 1993 [running]:
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile.func1()
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/executor/compiler.go:57 +0x360
panic({0x104246040?, 0x1400a1476f0?})
	/opt/homebrew/opt/go/libexec/src/runtime/panic.go:914 +0x218
github.com/pingcap/tidb/pkg/util/intest.doPanic(...)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/util/intest/assert.go:59
github.com/pingcap/tidb/pkg/util/intest.Assert(...)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/util/intest/assert.go:27
github.com/pingcap/tidb/pkg/expression/context.AssertLocationWithSessionVars(0x1046baa00?, 0x1400ae80000)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/context/context.go:232 +0x1c8
github.com/pingcap/tidb/pkg/expression/contextsession.(*SessionEvalContext).TypeCtx(0x1400a226240)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/contextsession/sessionctx.go:209 +0xb0
github.com/pingcap/tidb/pkg/expression.typeCtx(...)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/context.go:57
github.com/pingcap/tidb/pkg/expression.(*Constant).Equal(0x1400a17bec0, {0x104bf9788, 0x1400a226240}, {0x104c35ce0?, 0x1400a17bec0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/constant.go:437 +0xa0
github.com/pingcap/tidb/pkg/expression.(*baseBuiltinFunc).equal(0x1400a226480, {0x104bf9788, 0x1400a226240}, {0x104c0fb68?, 0x1400a226480?})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/builtin.go:400 +0xd8
github.com/pingcap/tidb/pkg/expression.(*ScalarFunction).Equal(0x1400a224460, {0x104bf9788, 0x1400a226240}, {0x104c35fb0?, 0x1400a224460})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/expression/scalar_function.go:372 +0xc4
github.com/pingcap/tidb/pkg/planner/cardinality.getMaskAndRanges({0x104bfedc0, 0x1400a147220}, {0x1400a147670, 0x1, 0x1}, 0x0?, {0x0?, 0x104b6e6e8?, 0x14006284fa8?}, 0x10022ce6c?, ...)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/cardinality/selectivity.go:730 +0x3cc
github.com/pingcap/tidb/pkg/planner/cardinality.Selectivity({0x104bfedc0, 0x1400a147220?}, 0x1400a223260, {0x1400a1475d0, 0x1, 0x1}, {0x1400a1472f0, 0x1, 0x104be0f80?})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/cardinality/selectivity.go:130 +0x970
github.com/pingcap/tidb/pkg/planner/core.(*DataSource).deriveStatsByFilter(0x140099d58c0, {0x1400a1475d0, 0x1, 0x1}, {0x1400a1472f0, 0x1, 0x2})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/stats.go:197 +0xe4
github.com/pingcap/tidb/pkg/planner/core.(*DataSource).DeriveStats(0x140099d58c0, {0x140099d58c0?, 0x400a0b2ff0?, 0x1?}, 0x14006285ee8?, {0x10022c584?, 0x1?, 0x1400a162528?}, {0x0?, 0x0?, ...})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/logical_datasource.go:328 +0x45c
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0x140099d58e0, {0x0, 0x0, 0x0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/operator/logicalop/base_logical_plan.go:188 +0x204
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0x1400a0b2ff0, {0x0, 0x0, 0x0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x114
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0x140099c3e20, {0x0, 0x0, 0x0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x114
github.com/pingcap/tidb/pkg/planner/core/operator/logicalop.(*BaseLogicalPlan).RecursiveDeriveStats(0x1400a0b2f20, {0x0, 0x0, 0x0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/operator/logicalop/base_logical_plan.go:181 +0x114
github.com/pingcap/tidb/pkg/planner/core.physicalOptimize({0x104c321d8, 0x1400a0b2f00}, 0x1400a2168a0)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/optimizer.go:1046 +0xdc
github.com/pingcap/tidb/pkg/planner/core.doOptimize({0x104bba9b8, 0x1400a1deb40}, {0x104bfedc0?, 0x1400a147220?}, 0x140062864c8?, {0x104c321d8, 0x1400a0b2f00})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/optimizer.go:308 +0xf0
github.com/pingcap/tidb/pkg/planner/core.DoOptimize({0x104bba9b8, 0x1400a1deb40}, {0x104bfedc0, 0x1400a147220}, 0x1400a224380?, {0x104c321d8, 0x1400a0b2f00})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/core/optimizer.go:355 +0xd0
github.com/pingcap/tidb/pkg/planner.optimize({0x104bba9b8, 0x1400a1deb40}, {0x104bfedc0?, 0x1400a147220}, {0x104bc6cf0?, 0x1400a0f8b40?}, {0x104c02388, 0x1400a0d17a0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/optimize.go:525 +0x500
github.com/pingcap/tidb/pkg/planner.Optimize({0x104bba9b8?, 0x1400a1deb40}, {0x104c39b10?, 0x140099ed900}, {0x104bc6cf0, 0x1400a0f8b40}, {0x104c02388?, 0x1400a0d17a0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/planner/optimize.go:356 +0x1178
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile(0x1400ae93840, {0x104bba9b8?, 0x1400a1deb40?}, {0x104bd1808?, 0x1400a0f8b40})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/executor/compiler.go:99 +0x2c8
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt(0x140099ed900, {0x104bba9b8?, 0x1400a1deb40?}, {0x104bd1808, 0x1400a0f8b40?})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/session/session.go:2098 +0x650
github.com/pingcap/tidb/pkg/session.(*session).ExecuteInternal(0x140099ed900, {0x104bba9b8, 0x1400a1deb40}, {0x1400a22bdc0, 0x3c}, {0x0, 0x0, 0x0})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/session/session.go:1523 +0x17c
github.com/pingcap/tidb/pkg/ddl/session.(*Session).Execute(0x1400a147230, {0x104bba9b8, 0x1400a1deb40}, {0x1400a22bdc0, 0x3c}, {0x103566932, 0xe})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/session/session.go:85 +0x150
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).GetMinJobID.func1(0x1400acbad98?)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/systable/manager.go:120 +0x94
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).withNewSession(0x1400acf2cd0, 0x14001488dd0)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/systable/manager.go:71 +0xc4
github.com/pingcap/tidb/pkg/ddl/systable.(*manager).GetMinJobID(0x1013ca433?, {0x104bba9b8?, 0x1400a1deb40?}, 0x8c?)
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/systable/manager.go:118 +0x58
github.com/pingcap/tidb/pkg/ddl/systable.(*MinJobIDRefresher).refresh(0x10359a57b?, {0x104bba9b8?, 0x1400a1deb40?})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/systable/min_job_id.go:66 +0x48
github.com/pingcap/tidb/pkg/ddl/systable.(*MinJobIDRefresher).Start(0x18?, {0x104bba9b8, 0x1400a1deb40})
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/systable/min_job_id.go:54 +0x38
github.com/pingcap/tidb/pkg/ddl.(*ddl).Start.func3()
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/ddl/ddl.go:869 +0x30
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run.func1()
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/util/wait_group_wrapper.go:157 +0x58
created by github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run in goroutine 47
	/Users/arenatlx/go/src/github.com/pingcap/tidb/pkg/util/wait_group_wrapper.go:155 +0x7c

```

### 2. What did you expect to see? (Required)
no panic

### 3. What did you see instead (Required)
occasionally panic

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #54868, close #54860

Problem Summary:

### What changed and how does it work?
as title, see this for root cause https://github.com/pingcap/tidb/issues/54868#issuecomment-2249663873
### Check List

Tests 

- [x] Unit test, only fix existing ones, should cover it
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

bootstrap with TiKV, test some ddl/dml, they should work and check no error logs.
bootstrap with unistore, same test as above.

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -1418,27 +1404,57 @@ func (do *Domain) Init(
 			return err
 		}
 	}
+	return nil
+}
+
+// Start starts the domain.
+func (do *Domain) Start() error {
```
Comment (lcwangchao) [inline pkg/domain/domain.go:1411]: How about renaming it to `StartBackgroundLoops` and adding some comments that some SQL can still be executed before domain started.

**Review by lcwangchao — APPROVED** [review]
Could you also fix the print message here?

https://github.com/pingcap/tidb/blob/7cdf9182753775cd2bf6302dfa1dfc48f1927a84/pkg/expression/context/context.go#L232-L234

to

```
 intest.Assert(ctxLocStr == varsLocStr && ctxLocStr == stmtLocStr, 
 	"location mismatch, ctxLoc: %s, varsLoc: %s, stmtLoc: %s", 
- 	ctxLoc.String(), ctxLocStr, stmtLocStr, 
+ 	ctxLocStr, varsLocStr, stmtLocStr,
```

File: pkg/domain/domain.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/domain/domain.go:1412]: https://github.com/pingcap/tidb/pull/54913/commits/0006970132ba77981c24bd66b868d869e2599c4f

i think `Start` is clear enough

**Comment by YangKeao** [comment]
Also, close #54860

**Review by zanmato1984 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**